docs/src/security/waf.md

```diff
@@ -0,0 +1,145 @@
+
+When a malicious request header is mistakenly included as part of a legitimate (victim) request to the site instead of two separate requests, it could circumvent that application's security methods and disclose sensitive information in the process. The malicious request usually attempts to exploit this end of message disagreement by including CRLF (carriage return and line feed) characters in the request header, which would typically be used to parse individual requests.
+
+{{< note title="Protection rules">}}
+The WAF monitors for requests that include a CRLF character or the word `http/\d`, in combination with an `HTTP` or `WEBDAV` method name, since the presence of these features indicate an attempt to inject a second request.
+{{< /note >}}
+
+### Header injection
+
+As a general class of security vulnerabilities, header injections occur when HTTP response headers are generated based on user input. An attacker is able to exploit the vulnerable cycle to include malicious content in an application's response headers to subsequent requests.
+
+{{< note title="Protection rules">}}
+The WAF monitors for header injection attempts that can occur via the payload and the header itself. Requests are blocked for those that contain the carriage return (CR; `%0d`) and line feed (LF; `%0a`) characters so that data is not returned in a response header and interpreted by the client, similar to the [response splitting](/security/waf.html#protection-rules-2) and [request smuggling](/security/waf.html#protection-rules) protection rules. 
+
+It will also monitor and detect newline characters in `GET` request argument values. 
+{{< /note >}}
+
+### Response splitting
+
+Response splitting is enabled by the presence of header injection vulnerabilities. The standard HTTP request and response cycle results in a single HTTP response returned to a user for each HTTP request they place on the server. An HTTP response splitting attack occurs when an attacker modifies the data included in the HTTP response header. That malicious data can then be returned to a user placing subsequent requests. 
+
+Depending on the characters added to the response header (such as line feeds and carriage returns), an attacker can create and return additional responses to legitimate user requests, hence the name response *splitting*. 
+
+{{< note title="Protection rules">}}
+The WAF monitors for requests that include the carriage return (CR; `%0d`) and line feed (LF; `%0a`) characters. The presence of these characters indicate an attempt to inject data in the response header and potentially force intermediate proxy servers to treat the message as two separate responses. 
+{{< /note >}}
+
+### HTTP Splitting
+
+[Need some help with intro description here. Seems like much of the above, only specific to filename POSTs. Or Rewrites, but that seems specific to Apache/PHP? Not really sure.]
```

Comment:
some description if we have one.

Reply:
(HTTP splitting) what are we protecting against when we monitor for \r and \n in REQUEST_FILENAME, and where is this variable?

Reply:
removed, not relevant for us.